pkg/runtime/cli.go

```diff
@@ -369,6 +369,13 @@ func FromFlags() (*DaprRuntime, error) {
 		runtimeConfig.AppHealthCheck = nil
 	}
 
+	if daprGlobalConfig.IsFeatureEnabled(globalConfig.Spec.Features, daprGlobalConfig.DynamicLoading) {
```

Comment:
Not a fan of this approach because config files, especially in self-hosted, are often put in `~/.dapr/config.yaml` and auto-loaded by all apps.

1. Do we absolutely need a feature flag? Can't we use a CLI flag instead, that is adding a new flag to `daprd` similar to `--watch-path <path>` (maybe as an alternative to `--components-path`, which then doesn't use hot-reload) or `--enable-component-reloading` (boolean) or whatever else?
2. If we absolutely need a feature flag, we shouldn't crash the runtime if components-path is unset. Just just a warning and don't enable hot-reloading. Because of what I said above, taht config files are often reused across many apps.